my-react-app/src/models/models.ts

```diff
@@ -0,0 +1,151 @@
+import { FieldError, UseFormRegisterReturn } from 'react-hook-form';
+
+export interface ProductModel {
+  id: number;
+  title: string;
+  price: number;
+  description: string;
+  category: string;
+  image: string;
+  rating: RatingModel;
+}
+
+export interface RatingModel {
+  rate: number;
+  count: number;
+}
+
+export interface ProductProps {
+  product: ProductModel;
+}
+
+export interface PageTitleProps {
+  setTitle: (namePage: string) => void;
+}
+
+export interface GeneralState {
+  pageTitle: string;
+}
+
+export interface FormInput {
+  id?: number;
+  title: string;
+  date: string;
+  country: string;
+  additional: boolean;
+  pay: string;
+  profilePicture: string;
+}
+
+export interface FormCardState {
+  title: string;
+  date: string;
+  country: string;
+  additional: boolean;
+  pay: string;
+  profilePictureUrl: FileList;
+}
+
+export interface FormCardInput {
+  card: FormInput;
+}
+
+export interface FormCards {
+  cards: FormInput[];
+}
+
+export type FormProps = {
+  addCard: (newCard: FormInput) => void;
+};
+
+export interface FormValidState {
+  inputTitleValid: boolean;
+  inputDateValid: boolean;
+  inputCountryValid: boolean;
+  inputPackageValid: boolean;
+  inputTransferValid: boolean;
+  inputDeliveryValid: boolean;
+  inputPayValid: boolean;
+  formValid: boolean;
+  inputPictureValid: boolean;
+  inputPictureUrl: string;
+}
+
+export interface InputDateInterface {
```

Comment:
consider adding generic interface
```ts
interface BaseRegister<FieldName extends string> {
  register: UseFormRegisterReturn<FieldName>;
  error?: FieldError;
}

```


```suggestion
export type InputDateInterface = BaseRegister<'date'>;
```